source/MaterialXView/Viewer.cpp

```diff
@@ -67,25 +67,22 @@ void writeTextFile(const std::string& text, const std::string& filePath)
     file.close();
 }
 
-mx::DocumentPtr loadLibraries(const mx::StringVec& libraryFolders, const mx::FileSearchPath& searchPath)
+mx::DocumentPtr loadLibraries(const mx::FilePathVec& libraryFolders, const mx::FileSearchPath& searchPath)
 {
     mx::DocumentPtr doc = mx::createDocument();
     for (const std::string& libraryFolder : libraryFolders)
     {
-        mx::FilePath path = searchPath.find(libraryFolder);
-        mx::FilePathVec filenames = path.getFilesInDirectory("mtlx");
-
         mx::CopyOptions copyOptions;
         copyOptions.skipConflictingElements = true;
 
         mx::XmlReadOptions readOptions;
         readOptions.skipConflictingElements = true;
-        for (const std::string& filename : filenames)
+
+        mx::FilePath libraryPath = searchPath.find(libraryFolder);
+        for (const mx::FilePath& filename : libraryPath.getFilesInDirectory("mtlx"))
         {
-            mx::FilePath file = path / filename;
             mx::DocumentPtr libDoc = mx::createDocument();
-            mx::readFromXmlFile(libDoc, file, mx::EMPTY_STRING, &readOptions);
-            libDoc->setSourceUri(file);
```

Comment:
I assume we don't need this anymore since import will add in a source URI?

Reply:
Yes, exactly.  It should be very rare that we need to explicitly set the source URI, and in this case the call to importLibrary() will handle that for us.